### ISSUE TITLE
[FR] Bug: rms_norm, silu_and_mul, rotary_embedding cause accuracy to drop to zero on mthreads (LiquidAI/LFM2.5-1.2B-Instruct)

### ISSUE THREAD

**Issue by Lxiparer** [FlagOpen/FlagGems#4821]
### Description

When running model inference on **LiquidAI/LFM2.5-1.2B-Instruct** with FlagGems enabled on Moore Threads MTT S5000, GPQA Diamond accuracy drops from 40.0% (FlagGems disabled) to **0%** (FlagGems enabled). The issue is traced to three specific operators: `rms_norm`, `silu_and_mul`, and `rotary_embedding`. Disabling these three operators fully restores accuracy to 40.0%.

### Environment

| Item | Value |
|------|-------|
| Hardware | mthreads |
| GPU | MTT S5000 × 8 |
| PyTorch | 2.7.1 |
| vLLM | 0.20.2 |
| vllm-plugin-fl | 0.2.0 |
| FlagGems | 5.0.0 |
| Triton Backend | mtgpu |

### Steps to Reproduce

1. Set up environment on mthreads (MTT S5000)
2. Install dependencies (PyTorch 2.7.1 + Triton + FlagGems 5.0.0 + vLLM 0.20.2 + vllm-plugin-fl)
3. Start vLLM inference service with FlagGems operators enabled:
   ```bash
   MUSA_VISIBLE_DEVICES=2 VLLM_PLUGINS=fl USE_FLAGGEMS=1 \
     VLLM_FL_PREFER_ENABLED=true \
     VLLM_FL_FLAGOS_WHITELIST="rms_norm,silu_and_mul,rotary_embedding" \
     vllm serve /data/models/LiquidAI/LFM2.5-1.2B-Instruct \
       --host 0.0.0.0 --port 8000 \
       --compilation-config '{"mode": 0}' \
       --max-model-len 32768 \
       --trust-remote-code
   ```
4. Run GPQA Diamond accuracy evaluation (50 questions, temperature=0, greedy decoding):
   ```bash
   python3 fast_gpqa.py --model LFM2.5-1.2B-Instruct --limit 50 --max-tokens 24576
   ```
5. Result: 0% (FlagGems disabled: 40.0%)

6. Disable the three operators (remove VLLM_FL_FLAGOS_WHITELIST or set empty), restart service, re-evaluate:
   - Result: 40.0% (fully recovered)

### Expected Behavior

FlagGems operators should maintain accuracy within 5% relative degradation of the native (FlagGems-disabled) baseline.

### Actual Behavior

| Configuration | GPQA Diamond Score |
|---|---|
| FlagGems disabled | 40.0% |
| FlagGems enabled (rms_norm + silu_and_mul + rotary_embedding) | **0%** |
| FlagGems enabled (above 3 operators disabled) | 40.0% |

The three operators together cause the model to produce completely incorrect outputs. Disabling any combination of fewer than all three does not fully resolve the issue — all three must be disabled simultaneously for accuracy recovery.

### Error Logs

```
/usr/local/lib/python3.10/dist-packages/triton/backends/mtgpu/compiler.py:391: RuntimeWarning: MUSA backend: num_stages>1 requested without using TME; falling back to num_stages=1.
  stages["ttgir"] = lambda src, metadata: self.make_ttgir(src, metadata, options, self.capability, self.warp_size)
```

Additional note: Triton JIT compilation is extremely slow on MUSA backend — each unique `seq_len` triggers full recompilation (~180 seconds per new shape). This causes the 50-question evaluation to exceed 3600s timeout with FlagGems enabled.

### FlagGems Integration Code

**Active operators (causing issue):**
- `rms_norm`
- `silu_and_mul`
- `rotary_embedding`

### Additional Context

- The issue appears to be specific to **mthreads** platform and may not reproduce on standard CUDA (NVIDIA) environments
- These three operators form the core forward-pass pipeline for LFM architecture (normalization → activation → position encoding), so their combined numerical drift cascades through all transformer layers
- `compilation_config.mode=0` is required because torch.compile crashes on MUSA (separate issue)
- Triton compilation slowness (~180s per unique seq_len) is a secondary issue but severely impacts debugging turnaround

### Possible Directions

- **rms_norm**: Check whether variance reduction uses FP32 accumulation on MUSA. If intermediate reduction is FP16, accumulated error cascades through deep transformer layers
- **rotary_embedding**: Check `sin`/`cos` trigonometric function precision on MUSA Triton backend, especially for large angle values at long sequence positions
- **silu_and_mul**: Check whether the fused `x * sigmoid(x)` kernel performs unnecessary type conversion before multiplication
- Run per-operator numerical comparison: `FlagGems kernel output` vs `torch.nn.functional` eager output with identical input tensors, measure max absolute error and relative error
- Investigate Triton mtgpu JIT compilation performance — per-seq_len recompilation indicates missing shape specialization caching
